### PR TITLE
[Issue 7711][pulsar-broker] Use original role instead of proxy role to check permissions

### LINKED ISSUE #7711: Broker doesn't use the original role to check if a client is allowed to consume or produce
**Describe the bug**

In the case when the [proxy roles](https://pulsar.apache.org/docs/en/security-authorization/#proxy-roles) are used to enable authentication between proxy and broker components, the **broker uses the proxy role instead of the original role** to check if it is allowed to consume or produce data from a topic. If the proxy role has more rights (consume and produce), then a client will be able to do both of these operations even if the original role says "only consume" or "only produce".

**To Reproduce**
1. Enable the proxy role
2. Allow the proxy role to consume and produce
3. Allow the test role to consume
4. Start producing messages with the test role
5. Messages are successfully published

**Expected behavior**
Some kind of exception is expected - not allowed operation.

**Screenshots**

**Desktop (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by vzhikserg**
Fixes #7711 

### Motivation

A client can have more permissions than its original role allows and it is a security violation.

### Modifications

Use the original role (if available) instead of the proxy role to check if a client is allowed to consume or produce messages.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/PulsarAuthorizationProvider.java [hunk 1]
```diff
@@ -557,9 +557,9 @@ private void validatePoliciesReadOnlyAccess() {
         switch (operation) {
             case LOOKUP: isAuthorizedFuture = canLookupAsync(topicName, role, authData);
```
Comment (jiazhai) [inline pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/PulsarAuthorizationProvider.java:558]: does lookup also need a change?

Comment (vzhikserg) [reply pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/PulsarAuthorizationProvider.java]: the lookup method was updated

**Comment by jiazhai** [comment]
@vzhikserg Thanks for the help.  Would you please help check whether the failed ci is related?
```
[ERROR] org.apache.pulsar.broker.admin.TopicBacklogQuotaTest.testRemoveBacklogQuota(org.apache.pulsar.broker.admin.TopicBacklogQuotaTest)
[INFO]   Run 1: PASS
[ERROR]   Run 2: TopicBacklogQuotaTest.testRemoveBacklogQuota:113 ? ServerSideError HTTP 500 In...
[INFO] 
[ERROR] org.apache.pulsar.broker.admin.TopicBacklogQuotaTest.testSetBacklogQuota(org.apache.pulsar.broker.admin.TopicBacklogQuotaTest)
[INFO]   Run 1: PASS
[ERROR]   Run 2: TopicBacklogQuotaTest.testSetBacklogQuota:92 ? ServerSideError HTTP 500 Intern...
```

**Pull request merged**